manager/state/raft/raft.go

```diff
@@ -1142,15 +1145,30 @@ func (n *Node) send(ctx context.Context, messages []raftpb.Message) error {
 			continue
 		}
 
+		ch := make(chan struct{})
```

Comment:
> I'm quite ignorant on this part of the code, but wouldn't it be easier to bucket messages by member then having sendToMember just take the list of messages and apply them in order?

The problem is dealing with multiple `sendToMember` calls. The first must finish before the second invokes any RPCs.

> Or encapsulate this in a struct (member?) so that each member has its own sending goroutine and a Send() function that just enqueues the message?

Yeah, that's the alternative I talked about in the PR description, but it's harder than it looks. We need to spawn the sending goroutine whenever we add a member and signal for it to terminate when that member is removed. We also need to shut down all the goroutines when raft shuts down. We need to do this in a way that doesn't interrupt sending queued messages on shutdown. There's a special case for members that we don't have open connections with yet, that would have to be split into a separate code path. And `Member` being poorly abstracted and in serious need of refactoring makes this harder still.

It's definitely possible, but when I tried to do it this way I found it was making the code messier and introducing more complexity than I wanted. I like that the approach in this PR is a small tweak.